platform/machine/aws/machine.go

```diff
@@ -85,9 +92,27 @@ func (am *machine) ConsoleOutput() string {
 	return am.console
 }
 
-func (am *machine) saveConsole() error {
-	var err error
-	am.console, err = am.cluster.api.GetConsoleOutput(am.ID(), true)
+func (am *machine) saveConsole(origConsole string) error {
+	// am.cluster.api.GetConsoleOutput(am.ID(), true) will loop until
+	// the returned console output is non-empty. However, if the
+	// instance has e.g. been running for several minutes, the returned
+	// output will be non-empty but won't necessarily include the most
+	// recent log messages.  So we loop until the post-termination logs
+	// are different from the pre-termination logs.
+	err := util.Retry(60, 5*time.Second, func() error {
```

Comment:
`GetConsoleOutput` can take up to 5 minutes. This should probably loop less agressively, or else network failure could cause kola to hang for a 5 hours.

I guess what we actually want is a `util.Retry` that retries until x amount of time has elapsed instead of N tries with a pause.

Reply:
We have that function already `util.WaitUntilReady`